File: apps/h5-app2/src/config/smartTts.ts
/**
 * 讯飞超拟人语音合成（WebSocket 直连）凭据。
 *
 * 浏览器在 WebSocket 握手时无法附加 `x-api-key` 头，因此直连必须使用
 * HMAC-SHA256 签名 URL（鉴权方式二）。此处凭据用于在浏览器端直接签名，
 * 已配置时不再请求后端 `/api/smart-tts/ws-url`。
 *
 * 注意：凭据会暴露给页面访问者，仅用于内网/演示环境。
 */
export const IFLYTEK_SMART_TTS = {
  appId: '52e4a842',
  apiKey: '0b93f27fe80262b9d6c1f17f38f18f27',
  apiSecret: 'ZTJiMDZmNGI0M2JkYWY2Nzk3NjhmYzFh',
  baseUrl: 'wss://cbm01.cn-huabei-1.xf-yun.com/v1/private/mcd9m97e6'
}

File: apps/ai-service/app/api/routes/smart_tts.py
"""iFlytek Super Smart TTS endpoint used by the browser-direct WebSocket flow.

Only ``GET /api/smart-tts/ws-url`` is exposed: it returns a signed WebSocket
URL the browser can use to connect to iFlytek directly (HMAC-SHA256, auth
method 2). The browser-side ``streamSmartTtsWs`` calls this once, then opens
the signed WebSocket itself, so no server-side streaming endpoint is needed.
"""

from fastapi import APIRouter, HTTPException

from app.api.deps import CurrentUser
from app.core.rbac import Principal
from app.services.smart_tts_service import smart_tts_service

router = APIRouter()


@router.get("/smart-tts/ws-url")
async def smart_tts_ws_url(principal: Principal = CurrentUser) -> dict[str, str]:
    """Return a signed WebSocket URL the browser can connect to directly.

    Browsers cannot attach an x-api-key header during the WS handshake, so
    direct connect uses an HMAC-SHA256 signed URL (auth method 2). Signing is
    done server-side: the frontend never holds iFlytek credentials, it calls
    this endpoint to get {url, app_id} and then opens the WebSocket itself.
    """
    if not smart_tts_service.configured:
        raise HTTPException(status_code=400, detail="iFlytek Super Smart TTS 未配置")
    if smart_tts_service.auth_method != 2:
        raise HTTPException(
            status_code=400,
            detail="浏览器直连需要 IFLYTEK_SMART_TTS_AUTH_METHOD=2（HMAC-SHA256 签名 URL）",
        )
    try:
        url = smart_tts_service.build_url()
    except RuntimeError as exc:
        raise HTTPException(status_code=400, detail=str(exc)) from exc
    return {"url": url, "app_id": smart_tts_service.app_id}
